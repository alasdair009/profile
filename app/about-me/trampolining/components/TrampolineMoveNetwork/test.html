<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
