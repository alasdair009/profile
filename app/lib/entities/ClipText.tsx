"use client";

import styled from "styled-components";
import {rem} from "polished";

export const ClipText = styled.span`
  align-items: center;
  //background: black;
  display: flex;
  font-size: ${rem(56)};
  height: 100%;
  justify-content: center;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  background: black;
  color: white;
`;