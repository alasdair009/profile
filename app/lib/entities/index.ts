"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
