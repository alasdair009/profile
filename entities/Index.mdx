import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
