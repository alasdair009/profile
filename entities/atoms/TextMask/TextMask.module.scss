@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";

.root {
  display: flex;
  height: 100%;
  width: 100%;

  &::before,
  &::after {
    background: var(--maskFill);
    content: "";
    flex: 1;
  }
}

.inner {
  display: flex;
  flex-direction: column;
  height: 100%;

  &::before,
  &::after {
    background: var(--maskFill);
    content: "";
    flex: 1;
  }
}

.rect {
  fill: var(--maskFill);
  mask: url(#mask);
}

.svg {
  height: 100%;
  width: 100%;
}

.text {
  alignment-baseline: middle;
  font-family: typography.$fontOpenSans;
  font-size: rem.toRem(map-get(typography.$fontSizes, large));
  font-weight: map-get(typography.$fontWeights, black);
  text-anchor: middle;
}
