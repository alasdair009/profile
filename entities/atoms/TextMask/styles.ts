import styled from "styled-components";
import { fonts, fontSizes, fontWeights } from "../../design-tokens/typography";
import { Property } from "csstype";

export const Root = styled.div<{ maskFill: Property.Fill }>`
  display: flex;
  height: 100%;
  width: 100%;

  &::before,
  &::after {
    background: ${({ maskFill }) => maskFill};
    content: "";
    flex: 1;
  }
`;

export const Rect = styled.rect<{ maskFill: Property.Fill }>`
  fill: ${({ maskFill }) => maskFill};
  mask: url(#mask);
`;

export const Text = styled.text`
  alignment-baseline: middle;
  font-family: ${fonts.body};
  font-size: ${fontSizes.large.rem};
  font-weight: ${fontWeights.black};
  text-anchor: middle;
`;
