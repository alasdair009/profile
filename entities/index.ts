export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
