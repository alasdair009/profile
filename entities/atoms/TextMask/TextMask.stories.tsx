import { TextMask } from "./TextMask";
import { Meta, StoryObj } from "@storybook/nextjs";
import { colors } from "@/entities";
import { expect, within } from "storybook/test";

const meta: Meta<typeof TextMask> = {
  component: TextMask,
  argTypes: {
    maskFill: {
      control: {
        type: "color",
        presetColors: [colors.whiteGhost, colors.blackEvil],
      },
    },
  },
  args: {
    maskFill: colors.whiteGhost,
    text: "hello",
  },
};
export default meta;

export const Default: StoryObj<typeof TextMask> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const textMaskElement = canvas.getByTestId(TextMask.name);

    await expect(textMaskElement).toBeInTheDocument();
    await expect(textMaskElement).toHaveTextContent("hello");
  },
};
