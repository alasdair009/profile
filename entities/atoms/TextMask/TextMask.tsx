import { Rect, Root, Text } from "./styles";
import { HTMLAttributes } from "react";
import { Property } from "csstype";

type TextMaskProps = {
  /**
   * Fill colour around the text.
   */
  maskFill: Property.Fill;
  /**
   * Text for the mask.
   */
  text: string;
} & HTMLAttributes<HTMLDivElement>;

/**
 * Mask formed from text to place over a background.
 */
export function TextMask({ maskFill, text, ...rest }: TextMaskProps) {
  return (
    <Root $maskFill={maskFill} data-testid={TextMask.name} {...rest}>
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100">
        <defs>
          <mask id="mask" x="0" y="0" width="100" height="100">
            <rect x="0" y="0" width="100" height="100" fill="white" />
            <Text x="50" y="50">
              {text}
            </Text>
          </mask>
        </defs>
        <Rect x="0" y="0" height="100" width="100" $maskFill={maskFill} />
      </svg>
    </Root>
  );
}
