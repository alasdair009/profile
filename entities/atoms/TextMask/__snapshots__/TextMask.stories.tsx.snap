// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/TextMask Default play-test 1`] = `
<div data-testid="TextMask"
     class="sc-ezTrPE dLWVaC"
>
  <svg xmlns="http://www.w3.org/2000/svg"
       viewbox="0 0 100 100"
  >
    <defs>
      <mask id="mask"
            x="0"
            y="0"
            width="100"
            height="100"
      >
        <rect x="0"
              y="0"
              width="100"
              height="100"
              fill="white"
        >
        </rect>
        <text x="50"
              y="50"
              class="sc-hDcvty kXMNnC"
        >
          hello
        </text>
      </mask>
    </defs>
    <rect x="0"
          y="0"
          height="100"
          width="100"
          class="sc-drnuxz fsrnSl"
    >
    </rect>
  </svg>
</div>
`;
